### PR TITLE
i2 board spec

### PR DESCRIPTION
**Pull request by AuroraTuring**
Add TDD for board spec file.

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  end
+
+  it "has consecutive placement" do
+    expect(@board.valid_placement?(@cruiser, ["B1", "C1", "D1"])).to be true #vertical
+    expect(@board.valid_placement?(@cruiser, ["B1", "B2", "B3"])).to be true #horizontal
+    expect(@board.valid_placement?(@cruiser, ["A1", "B2", "C3"])).to be false #diagonal
+    expect(@board.valid_placement?(@cruiser, ["A3", "A2", "A1"])).to be false #descending
+    expect(@board.valid_placement?(@cruiser, ["A1", "A2", "A4"])).to be false #not consecutive
+  end
+end
```
Comment (NeilTheSeal) [inline spec/board_spec.rb]: We need a test that the cells can be in backwards order as long as they are adjacent -
```
expect(@board.valid_placement?(@cruiser, ["A3", "A1", "A2"]).to be true
expect(@board.valid_placement?(@cruiser, ["C1", "A1", "B1"]).to be true
```

Comment (NeilTheSeal) [inline spec/board_spec.rb]: We also need a test that the cells cannot be diagonal -
```
expect(@board.valid_placement?(@cruiser, ["A1", "B2", "C3"]).to be false
```

File: spec/board_spec.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    it "knows if a ship is overlapping" do
+      @cell_1 = @board.cells["A1"]
+      @cell_2 = @board.cells["A2"]
+      @cell_3 = @board.cells["A3"]
+      @board.place(@cruiser, ["A1", "A2", "A3"])
+      expect(@board.valid_placement?(@submarine, ["A1", "B1"])).to be false
+    end
+  end
+
+
```
Comment (NeilTheSeal) [inline spec/board_spec.rb:108]: We are missing the tests for rendering - add this
```
  describe "#rendering" do
    it "renders an empty board" do
      render = @board.render
      render_true = @board.render(true)
      expected = "  1 2 3 4 \n" \
                 "A . . . . \n" \
                 "B . . . . \n" \
                 "C . . . . \n" \
                 "D . . . . \n"
      expect(render).to eq(expected)
      expect(render_true).to eq(expected)
    end
    it "renders a board with ships on it" do
      @board.place(@cruiser, %w[A1 A2 A3])
      render = @board.render
      render_true = @board.render(true)

      expected_opponent = "  1 2 3 4 \n" \
                          "A . . . . \n" \
                          "B . . . . \n" \
                          "C . . . . \n" \
                          "D . . . . \n"

      expected_self = "  1 2 3 4 \n" \
                      "A S S S . \n" \
                      "B . . . . \n" \
                      "C . . . . \n" \
                      "D . . . . \n"

      expect(render).to eq(expected_opponent)
      expect(render_true).to eq(expected_self)
    end
  end
```

**Comment by AuroraTuring** [comment]
The requested changes have been made!

**Pull request merged**